### ISSUE TITLE
Link processing is broken with changes from 2023-05-09

### ISSUE THREAD

**Issue by toxadx** [lervag/wiki.vim#305]
Latest changes (I guess from 2023-05-09) broke link navigation.
When I press Enter on a link I get the following:
```
Error detected while processing function wiki#link#follow[1]..wiki#link#get[3]..script /home/toxa/dotfiles/vim/.vim/plugged/wiki.vim/autoload/wiki/link/definitions.vim:
line   71:
E121: Undefined variable: wiki#link#def#reference
line   72:
E121: Undefined variable: a:link
E116: Invalid arguments for function matchstr(a:link.content, self.rx_target)
line   75:
E121: Undefined variable: a:link
E116: Invalid arguments for function searchpos('^\s*\[' . a:link.id . '\]: ', 'nW')[0]
line   76:
E121: Undefined variable: a:link
line   85:
E121: Undefined variable: a:link
E116: Invalid arguments for function getline(a:link.lnum_target)
line   86:
E121: Undefined variable: l:line
E116: Invalid arguments for function matchstr(l:line, g:wiki#rx#url)
line   87:
E121: Undefined variable: a:link
E116: Invalid arguments for function empty(a:link.url) | return | endif
line   92:
E133: :return not inside a function
line  188:
E171: Missing :endif
Error detected while processing function wiki#link#follow[1]..wiki#link#get:
line    3:
E121: Undefined variable: g:wiki#link#definitions#all
```

VIM version 9.0.1388

**Comment by lervag** [comment]
Yes, that's expected. See the latest [release notes](https://github.com/lervag/wiki.vim/releases/tag/v0.7) for an explanation/context.

Can you provide a simple example where it fails? Also, what is your relevant Vim configuration?

**Comment by lervag** [comment]
Ah, I can reproduce. But the bug is fixed in the `feat/handlers-and-resolvers` branch - I want to merge that one soon and it provides a lot of changes. Could you either 1) pin to the 0.7 tag which is safe, or 2) check out the feature branch and test it?

**Comment by lervag** [comment]
Pinning to the tag should not require any actions from you, so that should be safe. But when #304 is merged and until the next release, you may need to update your configuration. I believe #304 is the last major update, though.

**Comment by lervag** [comment]
I've merged #304 now, which fixes the issue. But please note that it does bring breaking changes.
